### DISCUSSION TITLE
Why interaction via gRPC and REST was chosen?

### DISCUSSION THREAD

**Discussion by Widlar** [fetchai/cosmpy#286] [Developers]
Why was interaction via gRPC and REST chosen?
As far as I know, Cosmos ecosystem has 3 APIs for interaction with nodes: 
1. legacy REST API 
2. gRPC
3. tendermint RPC

CosmPy uses both REST and gRPC interfaces. Whereas CosmJS uses Tendermint RPC. Won't this cause a problem? Since most of the endpoints in the chain-registry are oriented towards the RPC interface. In some cases, only one gRPC endpoint from chain-registry  worked.
I understand that gRPC is more compact in data transfer. My fear is that REST will be canceled and there will be few working endpoints on gRPC because many use CosmJS and its RPC. Keplr wallet also uses only RPC and REST interfaces.

**Comment by jrriehl** [comment]
Thanks for raising this discussion and sorry for the late response. The choices of REST and gRPC were made primarily for speed of initial development, and there is no particular reason for not supporting RPC other than it being a lower priority until now. That could change if there is significant support for adding RPC and we would certainly be happy for someone to create a PR for this.
